src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -84,16 +86,36 @@ public void resetTypeNameToCatCache() {
                   + " category '{}' is shadowed by '{}'.", typeCategory.getTypeName(),
                   typeCategory.getClass(), beforeRegCat.getClass());
             }
+            for (String deprecateName : typeCategory.getDeprecatedNames()) {
+              beforeRegCat = typeNameToCatCacheDeprecated.putIfAbsent(deprecateName, typeCategory);
+              if (beforeRegCat != null) {
+                LOGGER.warn("Page type category collision on deprecated category name '{}' the "
+                    + "colliding category '{}' is shadowed by '{}'.", deprecateName,
+                    typeCategory.getClass(), beforeRegCat.getClass());
+              }
+            }
           }
         }
       }
     }
+  }
+
+  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+    loadTypeNameToCategoryMap();
     return Collections.unmodifiableMap(typeNameToCatCache);
   }
 
+  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMapIncludeDeprecated() {
+    loadTypeNameToCategoryMap();
+    ConcurrentMap<String, IPageTypeCategoryRole> completeMap = typeNameToCatCacheDeprecated;
+    completeMap.putAll(typeNameToCatCache);
```

Comment:
using use getAllTypeNames instead of getDeprecatedNames: this readding over and over again is not needed.